### DISCUSSION TITLE
ZX Sizif reboots in 128k basic mode with Divmmc selected.

### DISCUSSION THREAD

**Discussion by oO-cozy-Oo** [UzixLS/zx-sizif-512#69] [Q&A]
I`ve never really used the 128k basic option (.128 and Divmmc enabled)to type in a program before, but I had a go and when I type in 10 PRINT"something" and press enter the Sizif crashes and reboots to 48k screen.I had a rev 1 board which I could use 128k with Divmmc.
When I disable the Divmmc and fo back into 128k mode all works fine?but them I cannot save to Divmmc as its not enabled?
Do I need to update the ROM or reflash the Altera FPGA? 
Cheers Cozy

**Comment by UzixLS** [comment]
This is expected behavior as esxdos/divmmc cannot correctly handle rom entry points in basic 128. There is two options how to outflank this:
1 - use patched basic 128
2 - disable divmmc entry points when basic128 is active
Second way I've implemented in firmware [v20230820](https://github.com/UzixLS/zx-sizif-512/releases/tag/fw-20230820), you can try to upgrade to it.

**Reply by AtomicRPM** [reply]
patched basic 128?  Is there one available somewhere?

**Reply by UzixLS** [reply]
You can google for Derby++ ROM, which have patched basic128 version
![изображение](https://github.com/user-attachments/assets/f85c50eb-ad8c-4f09-baa4-56a9642d5e55)

**Reply by AtomicRPM** [reply]
> You can google for Derby++ ROM, which have patched basic128 version ![изображение](https://private-user-images.githubusercontent.com/1556813/427866914-f85c50eb-ad8c-4f09-baa4-56a9642d5e55.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDMxNjk1ODYsIm5iZiI6MTc0MzE2OTI4NiwicGF0aCI6Ii8xNTU2ODEzLzQyNzg2NjkxNC1mODVjNTBlYi1hZDhjLTRmMDktYmFhNC01NmE5NjQyZDVlNTUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMjhUMTM0MTI2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MGI1NjU2YzUzMzE1NzM0NTVmYjRiODgwMTEzZDEzNDMyYTRmZDA3ZDdiNWUwMzlkZTkzYjg2ZjNkYTUxNGI4ZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Ysz2gCPmL97wRkNHYDyP8nFdgOOkD9nB7JO_-DTzwwU)

Thanks!

**Comment by oO-cozy-Oo** [comment]
Thank you for the reply Eugene Ill flash a new rom and try it out!! happy xmas and this is a fantastic pcb design keep them coming!!

**Comment by oO-cozy-Oo** [comment]
Well after a few months I`ve done it, I did reflash a new ROM with the latest FW but it didn`t work?? but then I realise I had to reflash the CPLD. I did get a cheap Blaster USB and relashed with the Programmer (Quartus Prime 24.1std) and now all works fine using the .128

Thanks again Eugene
